File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:

        maxi=float("-inf")
        curr=0
        for i in range(len(nums)):
            curr+=nums[i]
            maxi=max(maxi,curr)
            if curr<0:
                curr=0
        return maxi

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxsum=nums[0];
        int ans=nums[0];

        for(int i=1;i<nums.length;i++){
            maxsum=Math.max(nums[i],maxsum+nums[i]);
            ans = Math.max(ans,maxsum);
        }

        return ans;
    }
}